src/core_modules/capture-core/components/FormFields/New/HOC/withDataElementDescription.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import * as React from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { IconInfo16, Popover, colors, spacers } from '@dhis2/ui';
+import { withStyles } from '@material-ui/core/styles';
+
+const getStylesLabel = () => ({
+    iconContainer: {
+        paddingLeft: spacers.dp8,
+        '&:hover': {
+            cursor: 'pointer',
+        },
+    },
+    popOverContainer: {
+        padding: `${spacers.dp8} ${spacers.dp12}`,
+    },
+    label: {
+        fontWeight: 600,
+        paddingBottom: 4,
+    },
+});
+
+type State = {
+    popOverOpen: boolean
+}
+export const withDataElementDescription = () =>
+    (InnerComponent: React.ComponentType<any>) =>
+        withStyles(getStylesLabel)(class DataElementDescription extends React.Component<Object, State> {
+            iconRef: any;
+
+            constructor(props) {
+                super(props);
+                this.iconRef = React.createRef();
+                this.state = {
+                    popOverOpen: false,
+                };
+            }
+            renderDescription(description, url) {
+                const { classes } = this.props;
+                return (<div>
+                    <div
+                        role="button"
+                        tabIndex={-1}
+                        ref={this.iconRef}
+                        className={classes.iconContainer}
+                        onClick={() => { this.setState({ popOverOpen: true }); }}
+                    >
+                        <IconInfo16 color={colors.grey600} />
+                    </div>
+                    {this.state.popOverOpen &&
+                    <Popover
+                        reference={this.iconRef.current}
+                        onClickOutside={() => this.setState({ popOverOpen: false })}
+                    >
+                        { description ? <div className={classes.popOverContainer}>
+                            <div className={classes.label}>{i18n.t('Description')}</div>
+                            <div>{description}</div>
+                        </div> : null}
+                        { url ? <div className={classes.popOverContainer}>
+                            <div className={classes.label}>{i18n.t('URL')}</div>
+                            <a href={url} target={'_blank'} rel="noreferrer">{url}</a>
+                        </div> : null}
+                    </Popover>}
+                </div>);
+            }
+
+            render() {
+                const { dataElement, classes, ...passOnProps } = this.props;
```

Comment:
Might be of interest to others in the future, so writing my suggestion here (we started talking about this on Slack). We need to make some changes for this to work in every form. 

1. The data element should not be included in the input in this context (it is in the view event page currently, but shouldn't. In other forms it is missing). Therefore, we should include description and url directly like we do for the label (flat structure).
2. I think it makes sense to add the properties here: https://github.com/dhis2/capture-app/blob/e4a50693a337607969671e2599e4e95730b88d06/src/core_modules/capture-core/components/D2Form/field/configs/base/configBaseDefaultForm.js#L28. (Didn't check this thoroughly, just a hunch).

Great if you verify that this is working in the new single event form.

NIT: Maybe rename the file to 'withDescription'.

Reply:
thanks @JoakimSM for the suggestion. You are indeed right, the data element is not defined in some cases so I move to use the base props as suggested.